src/Sylius/Behat/Context/Ui/Shop/ProductContext.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Ui\Shop;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Page\Shop\Product\ShowPageInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class ProductContext implements Context
+{
+    /**
+     * @var ShowPageInterface
+     */
+    private $showPage;
+
+    /**
+     * @param ShowPageInterface $showPage
+     */
+    public function __construct(ShowPageInterface $showPage)
+    {
+        $this->showPage = $showPage;
+    }
+
+    /**
+     * @Then I should be able to access product :product
+     */
+    public function iShouldBeAbleToAccessProduct(ProductInterface $product)
+    {
+        $this->showPage->tryToOpen(['slug' => $product->getSlug()]);
+
+        Assert::true(
+            $this->showPage->isOpen(['slug' => $product->getSlug()]),
+            'Product show page should be open, but it does not.'
+        );
+    }
+
+    /**
+     * @Then I should not be able to access product :product
+     */
+    public function iShouldNotBeAbleToAccessProduct(ProductInterface $product)
+    {
+        $this->showPage->tryToOpen(['slug' => $product->getSlug()]);
+
+        Assert::false(
+            $this->showPage->isOpen(['slug' => $product->getSlug()]),
+            'Product show page should not be open, but it does.'
+        );
+    }
+
+    /**
+     * @When /^I open (this product) page$/
+     */
+    public function iOpenProductPage(ProductInterface $product)
+    {
+        $this->showPage->open(['slug' => $product->getSlug()]);
+    }
+
+    /**
+     * @Given I should see the product name :name
+     */
+    public function iShouldSeeProductName($name)
+    {
+        Assert::same(
```

Comment:
So, why do we use `::eq` at all? `Eq` is used 8 times in this PR, where `same` is used only once.